torch/csrc/jit/passes/bailout_graph.cpp

```diff
@@ -0,0 +1,203 @@
+#include <torch/csrc/jit/passes/alias_analysis.h>
+#include <torch/csrc/jit/passes/bailout_graph.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <memory>
+#include <unordered_set>
+
+namespace torch {
+namespace jit {
+
+struct BailOutGraphBuilderForNode {
+  BailOutGraphBuilderForNode(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)) {}
+
+  Value* addNewInputForValue(Value* old_value, Value* new_value = nullptr) {
+    auto node = old_value->node();
+    // this reduces the number of inputs to a bailout graph significantly
+    // making it easier to debug
+    if (node->kind() == prim::Constant) {
+      auto new_const = copy_graph_->createClone(node, {nullptr});
+      copy_graph_->block()->appendNode(new_const);
+      return new_const->output();
+    }
+
+    if (!new_value) {
+      new_value = copy_graph_->block()->addInput();
+      live_inputs_.push_back(old_value);
+    }
+    this->old_to_new_[old_value] = new_value;
+    new_value->copyMetadata(old_value);
+    return new_value;
+  }
+
+  void buildBailOutBlockFrom(Node* n) {
+    auto outer_node = n->owningBlock()->owningNode();
+    auto* block = copy_graph_->block();
+    if (n->kind() == prim::Loop) {
+      auto new_max_count = addNewInputForValue(n->inputs()[0]);
+      auto cur_iter = addNewInputForValue(n->blocks()[0]->inputs()[0]);
+      // subtract the number of iterations we already did
+      auto updated_max_trip_count = copy_graph_->create(aten::sub);
```

Comment:
use `insert` not `create`,

Reply:
If you use `create`, it skips all the schema checking that makes sure the op you are inserting is valid. It is fine to use a guard for a single element.

Reply:
will change